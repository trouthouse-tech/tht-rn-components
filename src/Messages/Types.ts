import {GestureResponderEvent} from 'react-native';

type BaseProps = {
  name: string;
  date: Date;
  status: MessageStatusEnum;
  image?: string;
};

type BaseTogglerProps = {
  onPress: (selectedMessageStatus: MessageType) => void;
};

export enum MessageStatusEnum {
  Scheduled = 'Scheduled',
  Canceled = 'Canceled',
  Completed = 'Completed',
}

export enum MessageType {
  Sessions = 'Sessions',
  Requests = 'Requests',
}

export enum ScheduledMessageType {
  Scheduled = 'Scheduled',
  History = 'History',
}

//the first component type props
export type MessageItemProps = BaseProps & {
  messageType: MessageType;
};

export type MessageListProps = {
  messages: MessageItemProps[];
};

export type MessageTogglerProps = BaseTogglerProps & {
  currentMessageType: MessageType;
};

//the second component type props
export type ScheduledMessageItemProps = BaseProps & {
  notes: string;
  durationInHours: number;
  messageType: ScheduledMessageType;
};

export type ScheduledMessageListProps = {
  messages: ScheduledMessageItemProps[];
};
