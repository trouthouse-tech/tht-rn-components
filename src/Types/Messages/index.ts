type BaseMessageProps = {
  name: string;
  date: Date;
  status: MessageStatusEnum;
  image?: string;
};

export enum MessageStatusEnum {
  Scheduled = 'Scheduled',
  Canceled = 'Canceled',
  Completed = 'Completed',
}

//the first component type props
export enum MessageTypeEnum {
  Sessions = 'Sessions',
  Requests = 'Requests',
}

export type MessageItemProps = BaseMessageProps & {
  messageType: MessageTypeEnum;
};

export type MessageListProps = {
  messages: MessageItemProps[];
};

export type MessageTogglerProps = {
  currentMessageType: MessageTypeEnum;
  onPress: (selectedMessageStatus: MessageTypeEnum) => void;
};

//the second component type props
export enum SessionMessageTypeEnum {
  Scheduled = 'Scheduled',
  History = 'History',
}

export type SessionMessageItemProps = BaseMessageProps & {
  notes: string;
  durationInHours: number;
  messageType: SessionMessageTypeEnum;
};

export type SessionMessageListProps = {
  messages: SessionMessageItemProps[];
};

export type SessionTogglerProps = {
  currentMessageType: SessionMessageTypeEnum;
  onPress: (selectedMessageStatus: SessionMessageTypeEnum) => void;
};
