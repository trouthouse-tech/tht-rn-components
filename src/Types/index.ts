import * as MessageTypes from './Messages/';

export {MessageTypes};
