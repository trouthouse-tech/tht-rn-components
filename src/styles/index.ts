import * as Colors from './color';
import * as Fonts from './font';

export {Colors, Fonts};
