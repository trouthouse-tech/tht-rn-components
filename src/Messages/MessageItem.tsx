import React from 'react';
import moment from 'moment';

import {View, Text, Image, StyleSheet, Dimensions} from 'react-native';

import {MessageItemProps, MessageStatusEnum} from './Types';
import {Colors} from '../styles/Color';

export const MessageItem: React.FC<MessageItemProps> = (props) => {
  const {
    name,
    date,
    messageType,
    status,
    image = 'https://via.placeholder.com/150',
  } = props;
  return (
    <View style={styles.messageStyle}>
      <View>
        <Text style={styles.nameTextStyle}>{name}</Text>

        {status !== MessageStatusEnum.Completed ? (
          <View style={styles.dateTextViewStyle}>
            <Text style={styles.dateDisplayTextStyle}>
              {moment(date).format('DD/MM/YY')}
            </Text>
            <Text style={styles.dateDisplayTextStyle}>
              {moment(date).format('h:mm a')}
            </Text>
          </View>
        ) : (
          <Text style={styles.statusDisplayTextStyle}>{status}</Text>
        )}
        <Text>{messageType}</Text>
      </View>
      <Image source={{uri: image}} style={styles.messageImgStyle} />
    </View>
  );
};

const deviceHeight = Dimensions.get('window').height;
const styles = StyleSheet.create({
  messageStyle: {
    borderWidth: StyleSheet.hairlineWidth,
    borderRadius: 15,
    padding: 20,
    marginVertical: 10,
    justifyContent: 'space-between',
    flex: 1,
    flexDirection: 'row',
    alignContent: 'center',
    alignItems: 'center',
  },
  nameTextStyle: {
    color: Colors.Black,
    fontSize: 20,
    fontWeight: '700',
  },
  dateTextViewStyle: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    paddingVertical: 4,
  },
  dateDisplayTextStyle: {
    paddingHorizontal: 3,
    fontSize: 16,
    fontWeight: '600',
  },
  statusDisplayTextStyle: {
    fontSize: 16,
    fontWeight: '600',
  },
  messageImgStyle: {
    height: deviceHeight / 7,
    width: deviceHeight / 7,
    borderRadius: 200,
  },
});
