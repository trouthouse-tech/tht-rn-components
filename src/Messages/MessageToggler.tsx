import React from 'react';

import {View, Text, StyleSheet, TouchableWithoutFeedback} from 'react-native';

import {MessageTogglerProps, MessageType} from './Types';
import {Colors} from '../styles/';

export const MessageToggler: React.FC<MessageTogglerProps> = (props) => {
  const {currentMessageType, onPress} = props;

  return (
    <View>
      {Object.values(MessageType).map((key: MessageType, index: number) => (
        <TouchableWithoutFeedback
          key={index}
          onPress={() => {
            onPress(key);
          }}>
          <Text
            style={
              currentMessageType === key
                ? styles.selectedTabStyle
                : styles.tabStyle
            }>
            {MessageType[key]}
          </Text>
        </TouchableWithoutFeedback>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  selectedTabStyle: {
    fontSize: 15,
    textDecorationLine: 'underline',
  },
  tabStyle: {
    fontSize: 15,
    color: Colors.Grey,
  },
});
