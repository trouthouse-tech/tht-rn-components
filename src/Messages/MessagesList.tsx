import React from 'react';

import {View, FlatList, StyleSheet} from 'react-native';

import {MessageTypes} from '../Types/';
import {MessageItem} from './MessageItem';

export const MessageList: React.FC<MessageTypes.MessageListProps> = ({
  messages,
}) => {
  return (
    <View style={styles.listStyle}>
      <FlatList
        keyExtractor={(item: MessageTypes.MessageItemProps, index: number) =>
          index.toString()
        }
        data={messages}
        renderItem={({item}: {item: MessageTypes.MessageItemProps}) => (
          <MessageItem
            name={item.name}
            status={item.status}
            messageType={item.messageType}
            date={item.date}
          />
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  listStyle: {
    marginVertical: 10,
  },
});
