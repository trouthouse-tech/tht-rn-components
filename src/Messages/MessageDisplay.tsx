import React from 'react';

import {View, Text} from 'react-native';

import {MessageListProps, MessageType} from './Types';
import {MessageList} from './MessagesList';
import {MessageToggler} from './MessageToggler';

export const MessageDisplay: React.FC<MessageListProps> = ({messages}) => {
  const [currentMessageStatus, setcurrentMessageStatus] = React.useState<
    MessageType
  >(MessageType.Sessions);

  const toggleMessageStatus = (selectedMessageStatus: MessageType) => {
    console.log(selectedMessageStatus);
    setcurrentMessageStatus(selectedMessageStatus);
  };

  return (
    <View>
      <MessageToggler
        onPress={(selectedMessageStatus: MessageType) => {
          return toggleMessageStatus(selectedMessageStatus);
        }}
        currentMessageType={currentMessageStatus}
      />
      <MessageList
        messages={messages.filter(
          (m) => m.messageType === currentMessageStatus
        )}
      />
      <Text>{currentMessageStatus}</Text>
    </View>
  );
};
