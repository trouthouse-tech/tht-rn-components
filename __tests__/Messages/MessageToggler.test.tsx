import {render, cleanup} from 'react-native-testing-library';
import React from 'react';

import {MessageToggler} from '../../src/Messages/MessageToggler';
import {MessageTypes} from '../../src/Types/';

afterEach(cleanup);

describe('<MessageToggler />', () => {
  it('should match snapshot', () => {
    const rendered = render(
      <MessageToggler
        currentMessageType={MessageTypes.MessageTypeEnum.Requests}
        onPress={(item) => {
          console.log(item);
        }}
      />
    ).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
