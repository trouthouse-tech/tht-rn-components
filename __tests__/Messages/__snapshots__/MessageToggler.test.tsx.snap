// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageToggler /> should match snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#C5CCD6",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginHorizontal": 5,
      "marginTop": 5,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "color": "#D3D3D3",
        "flex": 1,
        "fontSize": 18,
        "opacity": 1,
        "paddingBottom": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 20,
          },
          null,
        ]
      }
    >
      Sessions
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#000000",
        "borderBottomWidth": 3,
        "color": "#D3D3D3",
        "flex": 1,
        "fontSize": 18,
        "opacity": 1,
        "paddingBottom": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 20,
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      Requests
    </Text>
  </View>
</View>
`;
